src/librustc_yk_sections/mir_cfg.rs

```diff
@@ -243,40 +249,43 @@ impl<'a, 'tcx> PhiInserter<'a, 'tcx> {
     /// Insert PHI nodes, returning the mutated pack.
     fn pack(mut self) -> ykpack::Pack {
         let mut df = DominatorFrontiers::new(self.mir);
+        let num_tir_vars = self.def_sites.len();
+        let num_tir_blks = self.mir.basic_blocks().len(); // There's one TIR block per MIR block.
 
         // We first need a mapping from block to the variables it defines. Appel calls this
         // `A_{orig}`. We can derive this from our definition sites.
-        let (a_orig, num_blks) = {
+        let a_orig = {
             let ykpack::Pack::Mir(ykpack::Mir{ref blocks, ..}) = self.pack;
-            let mut a_orig: IndexVec<BasicBlock, FxHashSet<TirLocal>> =
-                IndexVec::with_capacity(blocks.len());
-            a_orig.resize(blocks.len(), FxHashSet::default());
+            debug_assert!(num_tir_blks == blocks.len());
+
+            let mut a_orig: IndexVec<BasicBlock, BitSet<TirLocal>> =
```

Comment:
If you want maximum performance (but maybe this is over-optimising...), you want a single bitset of `num_tir_blks * num_tir_vars`, skipping the vec entirely.

Reply:
Ah yeah, it's possible. Shall we put a comment and come back to it if we find we need it?

Reply:
Actually, wait, I'm not sure I follow. The vector is acting as a mapping between basic block index and a collection of TIR variables.

I think you are suggesting we could have a single: `BitSet<(BasicBlock, TirLocal)>`?

How would you look up the locals for any given basic block without iterating the whole set?

Either way, I think we can probably leave this as-is and optimise on-demand.

Reply:
Ah, I didn't explain things very well. Oops!

What I meant was having a BitVec of length `num_tir_blks * num_tir_vars`. Since all we're doing is setting / checking individual bits, this will be *very* fast.

Reply:
Can you show me the proposed type of `a_orig` and how you'd fetch the variables for (e.g.) block index 42?